examples/server/server.cpp

```diff
@@ -798,7 +798,7 @@ struct server_context {
         return last_used;
     }
 
-    bool launch_slot_with_data(server_slot & slot, json data) const {
+    bool launch_slot_with_data(server_slot & slot, json data, std::string & error_message) const {
```

Comment:
Instead of propagating an error message, change the signature to:

```suggestion
    bool launch_slot_with_task(server_slot & slot, const server_task & task) const {
```

And send errors via `send_error` from within the function